src/item.cpp

```diff
@@ -2535,6 +2535,27 @@ void item::magazine_info( std::vector<iteminfo> &info, const iteminfo_query *par
         return;
     }
 
+    if( parts->test( iteminfo_parts::MAGAZINE_COMPATIBLE_GUNS ) &&
+        islot_magazine::compatible_guns.find( this->typeId() ) != islot_magazine::compatible_guns.end() ) {
+        const auto compare = []( const auto & lhs, const auto & rhs ) {
+            return localized_compare( lhs, rhs );
+        };
+        std::set<std::string, decltype( compare )> compatible_guns( compare );
+
+        for( const itype_id &gun_type_id : islot_magazine::compatible_guns[this->typeId()] ) {
+            const itype &gun_type = gun_type_id.obj();
+            compatible_guns.insert( gun_type.nname( 1 ) );
+            for( const itype_variant_data &variant_type : gun_type.variants ) {
+                compatible_guns.insert( variant_type.alt_name.translated() );
+            }
```

Comment:
This will put both the generic version, which can never show up if you have variants on, and all the variants which will never appear if you have variants on.

Reply:
Made a change in https://github.com/CleverRaven/Cataclysm-DDA/pull/59728/commits/62e8d5b3068d47f84fa64c5d12747a0ca03ba1a2 , I checked with the toggle on and off and it seems to be working now, does that look better now?